ironfish/src/rpc/server.ts

```diff
@@ -86,30 +94,51 @@ export class RpcServer {
   /** Authenticate the RPC request */
   authenticate(requestAuthToken: string | undefined | null): boolean {
     if (!requestAuthToken) {
-      this.logger.debug(`Missing Auth token in RPC request.`)
       return false
     }
 
-    const rpcAuthToken = this.internal.get('rpcAuthToken')
+    if (!this.authTokenBuffer.byteLength) {
+      return false
+    }
 
-    if (!rpcAuthToken) {
-      this.logger.debug(`Missing RPC Auth token in internal.json config.`)
+    if (requestAuthToken.length >= AUTH_MAX_LENGTH) {
       return false
     }
 
-    return requestAuthToken === rpcAuthToken
+    const requestAuthBuffer = Buffer.alloc(AUTH_MAX_LENGTH)
+    requestAuthBuffer.write(requestAuthToken)
+    return timingSafeEqual(requestAuthBuffer, this.authTokenBuffer)
   }
 
   private async generateAuth(): Promise<void> {
     const rpcAuthToken = this.internal.get('rpcAuthToken')
 
     if (rpcAuthToken) {
-      this.logger.debug(
-        `Missing RPC Auth token in internal.json config. Automatically generating auth token.`,
-      )
-      const newPassword = randomBytes(32).toString('hex')
-      this.internal.set('rpcAuthToken', newPassword)
-      await this.internal.save()
+      this.loadAuth(this.internal.get('rpcAuthToken'))
+      return
     }
+
+    this.logger.debug(
+      `Missing RPC Auth token in internal.json config. Automatically generating auth token.`,
+    )
+    const newPassword = randomBytes(AUTH_MAX_LENGTH).toString('hex')
```

Comment:
This will produce a string that is `2 * AUTH_MAX_LENGTH` bytes long. Passing this string to `authenticate()` won't be possible because it will fail the length check. This should perhaps be:

```
randomBytes(AUTH_MAX_LENGTH / 2).toString('hex')
```

Reply:
Fixed.